packages/api/es/indexer.js

```diff
@@ -153,6 +153,20 @@ function indexExecution(esClient, payload, index = defaultIndexAlias, type = 'ex
   return genericRecordUpdate(esClient, payload.arn, payload, index, type);
 }
 
+/**
+ * Indexes a step function message to Elastic Search. The message must
+ * comply with the cumulus message protocol
+ *
+ * @param  {Object} esClient - ElasticSearch Connection object
+ * @param  {Object} payload  - Cumulus Step Function message
+ * @param  {string} index    - Elasticsearch index alias (default defined in search.js)
+ * @param  {string} type     - Elasticsearch type (default: asyncOperation)
+ * @returns {Promise} elasticsearch update response
+ */
+function indexAsyncOperation(esClient, payload, index = defaultIndexAlias, type = 'asyncOperation') {
+  return genericRecordUpdate(esClient, payload.arn, payload, index, type);
```

Comment:
I don't think the async operation is going to have an `arn` you'd probably need to go with the `asyncOperationId`